woocommerce/woocommerce-android
# "What's New" Screen: Add sample announcement and style it

This is step 3 for #4698 

## What this PR does
- Adds an example announcement,
- Adds proper view model and repository structure,
- Adds RecyclerView adapter to display the view model,
- Adds proper styling to the "What's New" dialog, both for phone and tablet layouts

## To test:
1. Run in a debug build,
2. Go to Settings, tap "What's New in WooCommerce"
3. Make sure the Dialog is shown properly, looking like below:

**Phone in portrait mode:**

| Light | Dark |
| - | - |
| ![Phone in portrait, light mode](https://user-images.githubusercontent.com/266376/133058169-6b351380-e0cf-453a-ad0e-a1c3ba037276.png) | ![Phone in portrait, dark mode](https://user-images.githubusercontent.com/266376/133058072-1be031a0-3b1c-493b-afdb-86011a226691.png) |

**Phone in landscape mode:**

| Light | Dark |
| - | - |
| ![Phone in landscape, light mode](https://user-images.githubusercontent.com/266376/133058376-1d989067-db4f-42ea-a134-1b3c180f90cc.png) | ![Phone in landscape, dark mode](https://user-images.githubusercontent.com/266376/133058459-842525dd-5e19-4121-81cb-3e51867a7c52.png) |

**Tablet in portrait mode:**

| Light | Dark |
| - | - |
| ![Tablet in portrait, light mode](https://user-images.githubusercontent.com/266376/133059648-5a92c913-5d79-4fd6-b684-a7bbd2f3ab59.png)| ![Tablet in portrait, dark mode](https://user-images.githubusercontent.com/266376/133059543-4488aa34-d65f-4315-944b-19ee19a65682.png) |


**Tablet in landscape mode:**

| Light | Dark |
| - | - |
| ![Tablet in landscape, light mode](https://user-images.githubusercontent.com/266376/133059102-b33bf2a1-f71e-40af-a365-462f20877c9c.png) | ![Tablet in landscape, dark mode](https://user-images.githubusercontent.com/266376/133059328-73d8f9b0-3247-4d70-bb4f-bba6f2d58aab.png) |
